services/repo_providers.py

```diff
@@ -86,8 +95,60 @@ def get_provider(service, adapter_params):
         raise TorngitInitializationFailed()
 
 
+def get_ghapp_default_installation(
+    owner: Optional[Owner],
+) -> Optional[GithubAppInstallation]:
+    if owner is None or owner.service not in [
+        Service.GITHUB.value,
+        Service.GITHUB_ENTERPRISE.value,
+    ]:
+        return None
+    return owner.github_app_installations.filter(
+        name=GITHUB_APP_INSTALLATION_DEFAULT_NAME
+    ).first()
+
+
+async def async_get_ghapp_default_installation(
```

Comment:
We often fix async/sync issues with the `@sync_to_async` or `@async_to_sync` decorators. Did you give that a shot as well? Curious if that would have fixed it

Reply:
Making `get_adapter` async didn't work cause it's the same issue (sync only op running in async loop)
Making `is_using_integration` async also didn't work (same as above)

I didn't try to make the interactor sync. I assumed it was async for a reason and decided not to touch it.

I did try some other helper functions with those wrappers but couldn't find a combination that worked. So I gave up and decided to split the function into two 🤷